src/newViews/PillarAnalysis/EntriesFilterForm/index.tsx

```diff
@@ -164,111 +173,104 @@ function EntriesFilterForm(props: OwnProps) {
             filter: data.properties,
             regions,
             geoOptions,
+            value,
+            onValueChange,
             className: _cs(
                 styles.filter,
                 isMatrixFilter && styles.showFilter,
             ),
         });
-    }, [regions, geoOptions]);
-
-    const handleFaramChange = useCallback((newValues) => {
-        setFaramValues(newValues);
-        setPristine(false);
-    }, []);
-
-    const handleFaramValidationSuccess = useCallback((_, finalValues) => {
-        onFiltersValueChange(finalValues);
-        setPristine(true);
-    }, [onFiltersValueChange]);
+    }, [regions, geoOptions, value, onValueChange]);
 
     const handleClearFilters = useCallback(() => {
-        setFaramValues({});
         onFiltersValueChange({});
-        setPristine(true);
     }, [onFiltersValueChange]);
 
     const pending = entryOptionsPending;
 
+    const handleSubmit = useCallback(() => {
+        onFiltersValueChange(value);
+    }, [onFiltersValueChange, value]);
+
     return (
-        <Faram
-            schema={schema}
-            value={faramValues}
-            error={faramErrors}
-            disabled={pending || disabled}
-            onValidationSuccess={handleFaramValidationSuccess}
-            onValidationFailure={setFaramErrors}
-            onChange={handleFaramChange}
+        <form
             className={_cs(
                 className,
                 styles.entriesFilterForm,
                 allFiltersVisible && styles.showFilters,
             )}
+            onSubmit={createSubmitHandler(validate, onErrorSet, handleSubmit)}
         >
             <MultiSelectInput
                 className={styles.filter}
-                faramElementName="created_by"
+                name="created_by"
                 keySelector={optionKeySelector}
                 labelSelector={optionLabelSelector}
+                value={value?.created_by as (string[] | undefined)}
```

Comment:
any reason we are using type assertions here? Shouldn't we define the types in FormType ?

Reply:
We don't really know types for all the auto-generated filter fields.
We'll need to change the data structure of widgets' filter in future to completely make this type safe.
For now, there is no other way.